Firestore/core/src/firebase/firestore/util/filesystem.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_FILESYSTEM_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_FILESYSTEM_H_
+
+#include "Firestore/core/src/firebase/firestore/util/path.h"
+#include "Firestore/core/src/firebase/firestore/util/status.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+// High-level routines for the manipulating the filesystem. All filesystems
+// are required to implement these routines.
+
+/**
+ * Returns true if the given path exists. If the path exists and is_directory is
+ * non-null, stores whether or not the path is a directory. If the file does
+ * not exist, is_directory is unmodified.
+ */
```

Comment:
Consider adding some text as to what happens during an error. (Most obviously permission denied. But other i/o errors may also prevent returning a valid answer.)

Possibly return a Status rather than a bool?

Reply:
Done.